File: cli/src/command/template/go/vorpal.go
package main

import (
	"log"

	api "github.com/ALT-F4-LLC/vorpal/sdk/go/pkg/api/artifact"
	"github.com/ALT-F4-LLC/vorpal/sdk/go/pkg/artifact/language"
	"github.com/ALT-F4-LLC/vorpal/sdk/go/pkg/config"
)

func main() {
	context := config.GetContext()

	systems := []api.ArtifactSystem{
		api.ArtifactSystem_AARCH64_DARWIN,
		api.ArtifactSystem_AARCH64_LINUX,
		api.ArtifactSystem_X8664_DARWIN,
		api.ArtifactSystem_X8664_LINUX,
	}

	_, err := language.NewGoBuilder("example", systems).
		WithIncludes([]string{"main.go", "go.mod", "go.sum"}).
		Build(context)
	if err != nil {
		log.Fatalf("error building: %v", err)
	}

	context.Run()
}

File: cli/src/command/template/rust/src/vorpal.rs
use anyhow::Result;
use vorpal_sdk::{
    artifact::{
        language::rust::{Rust, RustDevelopmentEnvironment},
        system::SYSTEMS,
    },
    context::get_context,
};

#[tokio::main]
async fn main() -> Result<()> {
    let mut ctx = get_context().await?;

    // -> 1. Activate: `source "$(vorpal build --path 'example-dev')/bin/activate"`
    // -> 2. Deactivate: `deactivate`

    RustDevelopmentEnvironment::new("example-dev", SYSTEMS)
        .build(&mut ctx)
        .await?;

    // -> 1. Build: `vorpal build 'example'`
    // -> 2. Run: `$(vorpal build --path 'example')/bin/example`

    Rust::new("example", SYSTEMS)
        .with_bins(vec!["example"])
        .with_includes(vec!["src/main.rs", "Cargo.lock", "Cargo.toml"])
        .build(&mut ctx)
        .await?;

    ctx.run().await
}
